test/elixir_analyzer/exercise_test/common_checks/private_helper_functions_test.exs

```diff
@@ -0,0 +1,452 @@
+              not power_pellet_active and touching_ghost
+            end
+
+            def win?(has_eaten_all_dots, power_pellet_active, touching_ghost) do
+              has_eaten_all_dots and not lose?(power_pellet_active, touching_ghost)
+            end
+          end
+        end
+
+      comment = %{
+        @comment
+        | params: %{actual: "defmacro super_and(a, b)", expected: "defmacrop super_and(a, b)"}
+      }
+
+      assert PrivateHelperFunctions.run(code, @pacman_stub) == [{:fail, comment}]
+    end
+
+    test "solution with public def" do
+      code =
+        quote do
+          defmodule Rules do
+            defguardp is_bool(bool) when bool == true or bool == false
+
+            defmacrop super_and(left, right) do
+              quote do: unquote(left) and unquote(right)
+            end
+
+            def do_or(left, right), do: left or right
+
+            def eat_ghost?(power_pellet_active, touching_ghost) when is_bool(touching_ghost) do
+              super_and(power_pellet_active, touching_ghost)
+            end
+
+            def score?(touching_power_pellet, touching_dot) do
+              do_or(touching_power_pellet, touching_dot)
+            end
+
+            def lose?(power_pellet_active, touching_ghost) do
+              not power_pellet_active and touching_ghost
+            end
+
+            def win?(has_eaten_all_dots, power_pellet_active, touching_ghost) do
+              has_eaten_all_dots and not lose?(power_pellet_active, touching_ghost)
+            end
+          end
+        end
+
+      comment = %{
+        @comment
+        | params: %{actual: "def do_or(a, b)", expected: "defp do_or(a, b)"}
+      }
+
+      assert PrivateHelperFunctions.run(code, @pacman_stub) == [{:fail, comment}]
+    end
+
+    test "only one comment will be shown" do
+      code =
+        quote do
+          defmodule Rules do
+            defguard is_bool(bool) when bool == true or bool == false
+
+            defmacro super_and(left, right) do
+              quote do: unquote(left) and unquote(right)
+            end
+
+            def do_or(left, right), do: left or right
+
+            def eat_ghost?(power_pellet_active, touching_ghost) when is_bool(touching_ghost) do
+              super_and(power_pellet_active, touching_ghost)
+            end
+
+            def score?(touching_power_pellet, touching_dot) do
+              do_or(touching_power_pellet, touching_dot)
+            end
+
+            def lose?(power_pellet_active, touching_ghost) do
+              not power_pellet_active and touching_ghost
+            end
+
+            def win?(has_eaten_all_dots, power_pellet_active, touching_ghost) do
+              has_eaten_all_dots and not lose?(power_pellet_active, touching_ghost)
+            end
+          end
+        end
+
+      comment = %{
+        @comment
+        | params: %{actual: "def do_or(a, b)", expected: "defp do_or(a, b)"}
+      }
```

Comment:
It would be better if the first time the rule is broken gets shown (`defguard`) instead of the last time, like we do in all of the naming convention checks for example.

Reply:
I will need to change the implementation then, now it's used in `MapSet`, which is unsorted. But yes, it's worth doing.